### DISCUSSION TITLE
Servo control for older combi heaters?

### DISCUSSION THREAD

**Discussion by huseyinozsut** [emsesp/EMS-ESP32#1556] [Ideas]
Hello. I own a Bosch condense comfort 35kw combi. This has a common controller card in several models. Controller knobs and screen in all of the models are same and looks like this:

![102398028](https://github.com/emsesp/EMS-ESP32/assets/73585157/3fde592b-8f16-45ba-b1cb-d69f5a4d7cf0)

This controller's heat knobs are analog. They don't turn infinitely. When knob is turned, it does not show the intended water heat; but only current heat. So, In EMS-ESP32 web screen, heat can't be changed; but only shown. Rest works perfectly.

So my solution would be to mount a servo motor to the heater knob. As heat increased in web screen the aimed water heat increased, servo motor can turn right, as it is decreased servo motor can turn left. I can do this with ESPHOME and add this to my HomeAssistant system; but since I allready have an ESP32 based controller near the Combi, I would prefer doing this with that and also would prefer to be able to change the heat level inside the EMS web screen.

Is this too hard to achieve?

**Comment by MichaelDvP** [comment]
My boiler also have mechanical knobs, 
![bc10-e1581426090307](https://github.com/emsesp/EMS-ESP32/assets/59284019/eb4469a8-bec9-4dce-97da-a91db2d67915)
but i can set the temperature via ems-esp to any value **below** the mechanical position. After 2 minutes the boiler resets to the knob position, you have to repeat the ems-setting command every minute.

**Reply by proddy** [reply]
I've made a note of this in the doc

**Reply by huseyinozsut** [reply]
When I looked again, you are right! :D Thank you. I guess, I will add a PWM output. Then I will customize the entitiy of water temp and link pwm with that somehow? Is it correct? I will try this in a few days. 

Thank you for all your affords by the way.

**Reply by huseyinozsut** [reply]
Or not bother to link the original water temp entitiy, but use the pwm output as a new entitiy. Practically the samething.

**Reply by MichaelDvP** [reply]
@proddy I think we schould change the API command for scheduler. We have the command as bool on/off, mainly for easy implementing the forceheatingoff in scheduler, but it's not used anymore. 
Better to set the value parameter of the scheduled command, maybe with optional id 0/1 for switching on/off. Then it's possible to periodical schedule the selflowtemp command every minute and change the value on demand.
What do you think?

**Reply by proddy** [reply]
I think we should keep the scheduler as a simple scheduler, i.e. being stupid and just calling commands, and then pick up #1423 where we can extend Custom Entities to also read/write from internal store (and not via an EMS telegram). The in the scheduler the Value could either be a hardcoded value or one of the Variables (like {var1}). It's a bit more work but really opens EMS-ESP to be more extendable.

**Reply by MichaelDvP** [reply]
Oh yes, you're right, i did not remember #1423.
